### PR TITLE
chore: use proper flaky test reporter index

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It uses the proper ES index

## Why is it important?
We do not want to mix-up flaky tests for different branches

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests for the CLI, and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [ ] @cachedout, we'd need to create the reporter and redeploy the pod

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -231,7 +231,7 @@ pipeline {
   post {
     cleanup {
       githubCheckNotify(currentBuild.currentResult == 'SUCCESS' ? 'SUCCESS' : 'FAILURE')
-      notifyBuildResult(analyzeFlakey: true, flakyReportIdx: "reporter-e2e-tests-end-2-end-tests-pipeline-master", prComment: true)
+      notifyBuildResult(analyzeFlakey: true, flakyReportIdx: "reporter-e2e-tests-end-2-end-tests-pipeline-7.10.x", prComment: true)
```
Comment (v1v) [inline .ci/Jenkinsfile:234]: What do you think to customise the index for branches or PRs, for branches we can use the BRANCH_NAME for PRs then we use the CHANGE_TARGET. (see https://beats-ci.elastic.co/pipeline-syntax/globals#env)

```suggestion
      whenTrue(isBranch()) {
         setEnvVar("SUFFIX", env.BRANCH_NAME))
      }
      whenTrue(isPR()) {
         setEnvVar("SUFFIX", env.CHANGE_TARGET))
      }
      notifyBuildResult(analyzeFlakey: (isBranch() || isPR()), flakyReportIdx: "reporter-e2e-tests-end-2-end-tests-pipeline-${SUFFIX}", prComment: true)
```

Comment (mdelapenya) [reply .ci/Jenkinsfile:234]: Would it be `${env.SUFFIX}`, or it doesn't matter?

Comment (mdelapenya) [reply .ci/Jenkinsfile:234]: @v1v I saw this old comment about the usage of the environment variable. Is it valid?

Comment (mdelapenya) [reply .ci/Jenkinsfile:234]: @v1v let's merge this PR as is, and create a _backportable_ PR with your suggestion. wdyt?

**Review by v1v — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Pull request merged**